scico/loss.py

```diff
@@ -202,7 +202,7 @@ def __init__(
             self.has_prox = True
 
     def __call__(self, x: Union[JaxArray, BlockArray]) -> float:
-        return self.scale * (self.W.diagonal * snp.abs(self.y - self.A(x)) ** 2).sum()
```

Comment:
This syntax is no longer possible? I guess that's OK as long as we clearly document what operations should be avoided to ensure array/blockarray compatibility.

Reply:
We could make this work, but it complicates the API. Currently all `BlockArray` methods map over blocks, including this one. You only get special reduction behavior when using `snp` methods.

Reply:
Probably not worth it then.